hummingbot/market/binance/binance_api_user_stream_data_source.py

```diff
@@ -104,32 +104,38 @@ async def get_ws_connection(self) -> websockets.WebSocketClientProtocol:
-            try:
-                if self._current_listen_key is None:
-                    self._current_listen_key = await self.get_listen_key()
-                    self.logger().debug(f"Obtained listen key {self._current_listen_key}.")
-                    if self._listen_for_user_stream_task is not None:
-                        self._listen_for_user_stream_task.cancel()
-                    self._listen_for_user_stream_task = safe_ensure_future(self.log_user_stream(output))
-                    await self.wait_til_next_tick(seconds=60.0)
-
-                success: bool = await self.ping_listen_key(self._current_listen_key)
-                if not success:
-                    self._current_listen_key = None
-                    if self._listen_for_user_stream_task is not None:
-                        self._listen_for_user_stream_task.cancel()
-                        self._listen_for_user_stream_task = None
-                    continue
-                self.logger().debug(f"Refreshed listen key {self._current_listen_key}.")
+        try:
+            while True:
+                try:
+                    if self._current_listen_key is None:
+                        self._current_listen_key = await self.get_listen_key()
+                        self.logger().debug(f"Obtained listen key {self._current_listen_key}.")
+                        if self._listen_for_user_stream_task is not None:
+                            self._listen_for_user_stream_task.cancel()
+                        self._listen_for_user_stream_task = safe_ensure_future(self.log_user_stream(output))
+                        await self.wait_til_next_tick(seconds=60.0)
+
+                    success: bool = await self.ping_listen_key(self._current_listen_key)
+                    if not success:
+                        self._current_listen_key = None
+                        if self._listen_for_user_stream_task is not None:
+                            self._listen_for_user_stream_task.cancel()
+                            self._listen_for_user_stream_task = None
+                        continue
+                    self.logger().debug(f"Refreshed listen key {self._current_listen_key}.")
 
-                await self.wait_til_next_tick(seconds=60.0)
-            except asyncio.CancelledError:
-                raise
-            except Exception:
-                self.logger().error("Unexpected error while maintaining the user event listen key. Retrying after "
-                                    "5 seconds...", exc_info=True)
-                await asyncio.sleep(5)
+                    await self.wait_til_next_tick(seconds=60.0)
+                except asyncio.CancelledError:
+                    raise
+                except Exception:
+                    self.logger().error("Unexpected error while maintaining the user event listen key. Retrying after "
+                                        "5 seconds...", exc_info=True)
+                    await asyncio.sleep(5)
```

Comment:
Is it possible for this `except` to cause infinite loop running when exception, other than CancelledError, happens?

Reply:
That is the intended behavior. Background tasks should keep running until cancelled. If it is any errors other than being cancelled, it should print error log instead of exit.